closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,12 +562,257 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         }
     }
 
+    if (traceEnabled) {
+        fprintf(stdout, "%s: %s\n", message, openssl_version);
+        fflush(stdout);
+    }
+
+    return ossl_ver;
+}
+
+static void *
+load_crypto_library_libname(jboolean traceEnabled, const char *libName)
+{
+    void *result = NULL;
+    if ((NULL != libName) && ('\0' != *libName)) {
+#if defined(_AIX)
+        int flags = RTLD_NOW;
+        if (NULL != strrchr(libName, '(')) {
+            flags |= RTLD_MEMBER;
+        }
+        result = dlopen(libName, flags);
+#elif defined (_WIN32)
+        result = LoadLibrary(libName);
+#else
+        result = dlopen(libName, RTLD_NOW);
+#endif
+    }
+    return result;
+}
+
+/* Load the crypto library (return NULL on error) */
+void * load_crypto_library(jboolean traceEnabled, const char *chomepath) {
+
+    void *result = NULL;
+    void *prevResult = NULL;
+    size_t numOfLibs = 0;
+    size_t i = 0;
+    long tempVersion = 0;
+    long previousVersion = 0;
+    size_t num_of_generic = 0;
+
+#if defined(_AIX)
+    num_of_generic = 4;
+#elif defined(__linux__)
+    num_of_generic = 1;
+#endif
+
+    /* Library names for OpenSSL 1.1.1, 1.1.0 and symbolic links
+     * It is important to preserve the order!!!
+     */
+    static const char * const libNames[] = {
+#if defined(_AIX)
+        "libcrypto.a(libcrypto64.so)",      /* general symlink library name from archive file */
+        "libcrypto64.so",                   /* general symlink library name */
+        "libcrypto.a(libcrypto.so)",        /* general symlink library name from archive file */
+        "libcrypto.so",                     /* general symlink library name */
+        "libcrypto.a(libcrypto64.so.3)",    /* 3.x library name from archive file */
+        "libcrypto64.so.3",                 /* 3.x library name */
+        "libcrypto.a(libcrypto.so.3)",      /* 3.x library name from archive file */
+        "libcrypto.so.3",                   /* 3.x library name */
+        "libcrypto.a(libcrypto64.so.1.1)",  /* 1.1.x library name from archive file */
+        "libcrypto.so.1.1",                 /* 1.1.x library name */
+        "libcrypto.so.1.0.0",               /* 1.0.x library name */
+#elif defined(__APPLE__)
+        "libcrypto.3.dylib",                /* 3.x library name */
+        "libcrypto.1.1.dylib",              /* 1.1.x library name */
+        "libcrypto.1.0.0.dylib",            /* 1.0.x library name */
+        /* "libcrypto.dylib"                Apple no longer supports loading symlink. */
```

Comment:
I'm not sure this comment is necessary.

Reply:
This generic library name is not supported by newer versions of Apple software, but we thought about keeping it there for now in case something breaks in older versions and we need to reinstate it.

Reply:
Removed.

Reply:
I've updated this comment to explain the reasoning behind commenting this out. Let me know if it makes sense.